From 79b463fffa41321cd001cc775548676731183b0a Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Fri, 3 Aug 2018 09:32:36 +0200
Subject: [PATCH] dts: 750_8215: move phys from davinci to marvell

After rework of interface board of 8215 Marvell Switch is
in single chip mode now. The PHYs are not connected to CPU
MDIO bus directly, but over Marvell switch nested MDIO bus.
This patch move part of PHY description into Marverll switch
block.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 arch/arm/boot/dts/am335x-pfc-750_8215.dts | 49 ++++++++++++++++++-------------
 1 file changed, 29 insertions(+), 20 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pfc-750_8215.dts b/arch/arm/boot/dts/am335x-pfc-750_8215.dts
index 8a777e3..57540ec 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8215.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8215.dts
@@ -105,26 +105,6 @@
 	pinctrl-1 = <&davinci_mdio_sleep_pins>;
 	status = "okay";
 
-	ext_phy0: ethernet-phy@0 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <0>;
-	};
-
-	ext_phy1: ethernet-phy@1 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-	};
-
-	int_phy3: ethernet-phy@3 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <3>;
-	};
-
-	int_phy4: ethernet-phy@4 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <4>;
-	};
-
 	switch: switch@0 {
 		compatible = "marvell,mv88e6085";
 		#address-cells = <1>;
@@ -134,6 +114,35 @@
 		reset-gpios = <&io_expander_70 4 GPIO_ACTIVE_LOW>;
 		phy-scan;
 
+		mdio {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			ext_phy0: ethernet-phy@0 {
+				compatible = "ethernet-phy-idD565.A401", "ethernet-phy-ieee802.3-c22";
+				reg = <0>;
+				max-speed = <100>;
+			};
+
+			ext_phy1: ethernet-phy@1 {
+				compatible = "ethernet-phy-idD565.A401", "ethernet-phy-ieee802.3-c22";
+				reg = <1>;
+				max-speed = <100>;
+			};
+
+			int_phy3: ethernet-phy@3 {
+				compatible = "ethernet-phy-id0141.0C00", "ethernet-phy-ieee802.3-c22";
+				reg = <3>;
+				max-speed = <100>;
+			};
+
+			int_phy4: ethernet-phy@4 {
+				compatible = "ethernet-phy-id0141.0C00", "ethernet-phy-ieee802.3-c22";
+				reg = <4>;
+				max-speed = <100>;
+			};
+		};
+
 		ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.7.4

